src/background/adblocker.js

```diff
@@ -196,25 +219,53 @@ async function injectScriptlets(scripts, tabId, frameId) {
     script.remove();
   }
 
-  chrome.scripting.executeScript(
-    {
-      injectImmediately: true,
-      world:
-        chrome.scripting.ExecutionWorld?.MAIN ??
-        (__PLATFORM__ === 'firefox' ? undefined : 'MAIN'),
-      target: {
-        tabId,
-        frameIds: [frameId],
-      },
-      func: scriptletInjector,
-      args: [encodeURIComponent(scriptlets)],
-    },
-    () => {
-      if (chrome.runtime.lastError) {
-        console.warn(chrome.runtime.lastError);
+  if (__PLATFORM__ === 'firefox') {
+    if (scripts.length === 0) {
+      contentScripts.delete(hostname);
```

Comment:
Why not put the `delete()` method logic here? The concept of `contentScripts` hides logic, but the unregister is related to this function.

Reply:
we do delete in mutliple places

Reply:
In one function. Instead of proxying map, just create two functions `clearContentScripts()` and `deleteContentScript(hostname)`, and "normal" at the module level `Map` to keep track of hostnames.

Reply:
what's the problem with this solution? this is just encapsulation

Reply:
It looks that this is some kind of map, but it does side effects in functions, which normally don't do that. It is not obvious that calling `contentScripts.delete()` it will also unregister, until you read the implementation. If you would create function `unregisterContentScripts()` (my new idea of naming), it might contain `map.delete(key)`, but the logic is explicit.

Reply:
but then someone could modify the map and forget to unregister the content script

Reply:
The current implementation looks much better.